server/src/transport/http.rs

```diff
@@ -157,41 +156,41 @@ where
 {
 	let Batch { data, call } = b;
 
-	if let Ok(batch) = serde_json::from_slice::<Vec<Request>>(&data) {
-		let max_response_size = call.max_response_body_size;
-		let batch = batch.into_iter().map(|req| Ok((req, call.clone())));
-
-		let batch_stream = futures_util::stream::iter(batch);
-
-		let batch_response = batch_stream
-			.try_fold(
-				BatchResponseBuilder::new_with_limit(max_response_size as usize),
-				|batch_response, (req, call)| async move {
-					let response = execute_call(req, call).await;
-					batch_response.append(&response)
-				},
-			)
-			.await;
-
-		return match batch_response {
-			Ok(batch) => batch.finish(),
-			Err(batch_err) => batch_err,
-		};
-	}
+	if let Ok(batch) = serde_json::from_slice::<Vec<&JsonRawValue>>(&data) {
+		let mut batch_response = BatchResponseBuilder::new_with_limit(call.max_response_body_size as usize);
+		let mut got_notif = false;
 
-	if let Ok(batch) = serde_json::from_slice::<Vec<Notif>>(&data) {
-		return if !batch.is_empty() {
-			BatchResponse { result: "".to_string(), success: true }
+		for val in batch {
+			if let Ok(req) = serde_json::from_str::<Request>(val.get()) {
+				let response = execute_call(req, call.clone()).await;
```

Comment:
NOTE: this is enforcing all futures to be processed in order which might reduce the throughput (if a heavy request blocks other ones)

I could change to use `stream.try_fold` again but then I need to change `Vec<Box<JsonRawValue>>` because of some trait bounds are not satisfied or use `FuturesUnordered` for the futures but that will likely cause the responses to be out-of-order all the time.

Reply:
I got to annoyed but that and decided to rework it a bit.

I need to bench it.